cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java

```diff
@@ -36,237 +36,1616 @@ public NGSICharsetsTest() {
+        
+        try {
+            assertEquals(expected, out);
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                    + "-  OK  - '" + in + "' has been encoded as '" + expected + "'");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                    + "- FAIL - '" + in + "' has not been encoded as '" + out + "' instead of '" + expected + "'");
+            throw e;
+        } // try catch
+    } // testEncodeMongoDBCollectionEquals
+    
+    /**
+     * [NGSICharsets.encodeMongoDBCollection] -------- A single 'x' is not encoded.
+     */
+    @Test
+    public void testEncodeMongoDBCollectionSinglex() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                + "-------- A single 'x' is not encoded");
+        String in = "x";
+        String expected = "x";
+        String out = NGSICharsets.encodeMongoDBCollection(in);
+        
+        try {
+            assertEquals(expected, out);
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                    + "-  OK  - '" + in + "' has not been encoded");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                    + "- FAIL - '" + in + "' has encoded as '" + out + "'");
+            throw e;
+        } // try catch
+    } // testEncodeMongoDBCollectionSinglex
+    
+    /**
+     * [NGSICharsets.encodeMongoDBCollection] -------- "xffff" is encoded (escaped) as "xxffff".
+     */
+    @Test
+    public void testEncodeMongoDBCollectionxffff() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                + "-------- \"xffff\" is encoded (escaped) as \"xxffff\"");
+        String in = "xffff";
+        String expected = "xxffff";
+        String out = NGSICharsets.encodeMongoDBCollection(in);
+        
+        try {
+            assertEquals(expected, out);
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                    + "-  OK  - '" + in + "' has been encoded as '" + expected + "'");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                    + "- FAIL - '" + in + "' has been encoded as '" + out + "' instead of '" + expected + "'");
+            throw e;
+        } // try catch
+    } // testEncodeMongoDBCollectionxffff
+    
+    /**
+     * [NGSICharsets.encodeMongoDBCollection] -------- Forbidden characters are encoded.
+     */
+    @Test
+    public void testEncodeMongoDBCollectionForbiddenCharacters() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMongoDBCollection]")
+                + "-------- Forbidde characters are encoded");
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4